### PR TITLE
[de] suggestions for wrong dates

### PR DESCRIPTION
**Pull request by jaumeortola**
Please look at the suggestions and the new message.  @Luisa-LT 

We could remove one dot here: 
`<example correction="Montag., 2015-09-28|Mi., 2015-09-30">Ein Termin am <marker>Mi., 2015-09-28</marker>.</example>`

### CODE REVIEW

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 1]
```diff
@@ -30502,15 +30502,15 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
-                <example correction="">Termin: <marker>Sonntag, 23.8.2014</marker></example>
-                <example correction="">Termin: <marker>So, 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>So., 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>Sonntag, den 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>So, den 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>So., den 23.08.2014</marker></example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>.08.2014</example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>.8.2014</example>
+                <example correction="Samstag, 23|So, 24">Termin: <marker>So, 23</marker>.08.2014</example>
+                <example correction="Samstag., 23|So., 24">Termin: <marker>So., 23</marker>.08.2014</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30510]: When the weekday is spelled out (Samstag), we don't need the dot (only the comma: Samstag,) but when the weekday is shortened (So.) we need the dot (and the comma: So.,)

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 2]
```diff
@@ -30502,15 +30502,15 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
-                <example correction="">Termin: <marker>Sonntag, 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>Sonntag, 23.8.2014</marker></example>
-                <example correction="">Termin: <marker>So, 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>So., 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>Sonntag, den 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>So, den 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>So., den 23.08.2014</marker></example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>.08.2014</example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>.8.2014</example>
+                <example correction="Samstag, 23|So, 24">Termin: <marker>So, 23</marker>.08.2014</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30509]: see comment below --> So.,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 3]
```diff
@@ -30502,15 +30502,15 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
+                <example correction="Samstag., 23|So., 24">Termin: <marker>So., 23</marker>.08.2014</example>
+                <example correction="Samstag, den 23|Sonntag, den 24">Termin: <marker>Sonntag, den 23</marker>.08.2014</example>
+                <example correction="Samstag, den 23|So, den 24">Termin: <marker>So, den 23</marker>.08.2014</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30512]: see comment above --> So.,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 4]
```diff
@@ -30502,15 +30502,15 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
+                <example correction="Samstag., den 23|So., den 24">Termin: <marker>So., den 23</marker>.08.2014</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30513]: see comment above -->Samstag,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 5]
```diff
@@ -30534,13 +30534,13 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
+                <filter class="org.languagetool.rules.de.DateCheckFilter" args="year:8 month:7 day:5 weekDay:1"/>
                 <message>Das Datum \5. \7 \8 fällt nicht auf einen {day}, sondern auf einen {realDay}.</message>
-                <example correction="">Termin: <marker>Donnerstag, 1. Jänner 2014</marker></example>
-                <example correction="">Termin: <marker>Sonntag, 23. August 2014</marker></example>
-                <example correction="">Termin: <marker>So, 23. August 2014</marker></example>
-                <example correction="">Termin: <marker>So., 23. August 2014</marker></example>
-                <example correction="">Termin: <marker>Sonntag, den 23. August 2014</marker></example>
+                <example correction="Mittwoch, 1|Donnerstag, 2">Termin: <marker>Donnerstag, 1</marker>. Jänner 2014</example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>. August 2014</example>
+                <example correction="Samstag, 23|So, 24">Termin: <marker>So, 23</marker>. August 2014</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30541]: -->So.,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 6]
```diff
@@ -30534,13 +30534,13 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
+                <example correction="Samstag., 23|So., 24">Termin: <marker>So., 23</marker>. August 2014</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30542]: --> Samstag,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 7]
```diff
@@ -30534,13 +30534,13 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
+                <example correction="Samstag, den 23|Sonntag, den 24">Termin: <marker>Sonntag, den 23</marker>. August 2014</example>
                 <example>Termin: <marker>Mittwoch, 1. Jänner 2014</marker></example>
                 <example>Termin: <marker>Samstag, 23. August 2014</marker></example>
                 <example>Termin: <marker>Sa, 23. August 2014</marker></example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30546]: --> Sa.,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 8]
```diff
@@ -30562,13 +30562,13 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
+                <filter class="org.languagetool.rules.de.DateCheckFilter" args="year:9 month:7 day:5 weekDay:1"/>
                 <message>Das Datum \5. \7 \9 fällt nicht auf einen {day}, sondern auf einen {realDay}.</message>
-                <example correction="">Termin: <marker>Donnerstag, 1. Jän. 2014</marker></example>
-                <example correction="">Termin: <marker>Sonntag, 23. Aug. 2014</marker></example>
-                <example correction="">Termin: <marker>So, 23. Aug. 2014</marker></example>
-                <example correction="">Termin: <marker>So., 23. Aug. 2014</marker></example>
-                <example correction="">Termin: <marker>Sonntag, den 23. Aug. 2014</marker></example>
+                <example correction="Mittwoch, 1|Donnerstag, 2">Termin: <marker>Donnerstag, 1</marker>. Jän. 2014</example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>. Aug. 2014</example>
+                <example correction="Samstag, 23|So, 24">Termin: <marker>So, 23</marker>. Aug. 2014</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30569]: -->So.,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 9]
```diff
@@ -30562,13 +30562,13 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
+                <example correction="Samstag., 23|So., 24">Termin: <marker>So., 23</marker>. Aug. 2014</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30570]: --> Samstag,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 10]
```diff
@@ -30562,13 +30562,13 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
+                <example correction="Samstag, den 23|Sonntag, den 24">Termin: <marker>Sonntag, den 23</marker>. Aug. 2014</example>
                 <example>Termin: <marker>Mittwoch, 1. Jän. 2014</marker></example>
                 <example>Termin: <marker>Samstag, 23. Aug. 2014</marker></example>
                 <example>Termin: <marker>Sa, 23. Aug. 2014</marker></example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30574]: --> Sa.,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 11]
```diff
@@ -30581,17 +30581,17 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
                 </pattern>
-                <filter class="org.languagetool.rules.de.YMDDateCheckFilter" args="weekDay:\1 date:\4"/>
+                <filter class="org.languagetool.rules.de.DateCheckFilter" args="weekDay:1 date:4"/>
                 <message>Das Datum \4 (JJJJ-MM-TT) fällt nicht auf einen {day}, sondern auf einen {realDay}.</message>
-                <example correction="">Ein Termin am <marker>Dienstag, 2015-09-28</marker>.</example>
-                <example correction="">Ein Termin am <marker>Mi, 2015-09-28</marker>.</example>
-                <example correction="">Ein Termin am <marker>Mi., 2015-09-28</marker>.</example>
+                <example correction="Montag, 2015-09-28|Dienstag, 2015-09-29">Ein Termin am <marker>Dienstag, 2015-09-28</marker>.</example>
+                <example correction="Montag, 2015-09-28|Mi, 2015-09-30">Ein Termin am <marker>Mi, 2015-09-28</marker>.</example>
+                <example correction="Montag., 2015-09-28|Mi., 2015-09-30">Ein Termin am <marker>Mi., 2015-09-28</marker>.</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30588]: --> Montag,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 12]
```diff
@@ -30581,17 +30581,17 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
                     <token regexp="yes">\d\d\d\d-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])</token>
                 </pattern>
-                <filter class="org.languagetool.rules.de.YMDDateCheckFilter" args="weekDay:\1 date:\4"/>
+                <filter class="org.languagetool.rules.de.DateCheckFilter" args="weekDay:1 date:4"/>
                 <message>Das Datum \4 (JJJJ-MM-TT) fällt nicht auf einen {day}, sondern auf einen {realDay}.</message>
-                <example correction="">Ein Termin am <marker>Dienstag, 2015-09-28</marker>.</example>
-                <example correction="">Ein Termin am <marker>Mi, 2015-09-28</marker>.</example>
-                <example correction="">Ein Termin am <marker>Mi., 2015-09-28</marker>.</example>
+                <example correction="Montag, 2015-09-28|Dienstag, 2015-09-29">Ein Termin am <marker>Dienstag, 2015-09-28</marker>.</example>
+                <example correction="Montag, 2015-09-28|Mi, 2015-09-30">Ein Termin am <marker>Mi, 2015-09-28</marker>.</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30587]: --> Mi.,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 13]
```diff
@@ -30677,15 +30677,15 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
-                <example correction="">Termin: <marker>Sonntag, 23.8.</marker></example>
-                <example correction="">Termin: <marker>So, 23.08.</marker></example>
-                <example correction="">Termin: <marker>So., 23.08.</marker></example>
-                <example correction="">Termin: <marker>Sonntag, den 23.08.</marker></example>
-                <example correction="">Termin: <marker>So, den 23.08.</marker></example>
-                <example correction="">Termin: <marker>So., den 23.08.</marker></example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>.08.</example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>.8.</example>
+                <example correction="Samstag, 23|So, 24">Termin: <marker>So, 23</marker>.08.</example>
+                <example correction="Samstag., 23|So., 24">Termin: <marker>So., 23</marker>.08.</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30685]: --> Samstag,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 14]
```diff
@@ -30677,15 +30677,15 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
-                <example correction="">Termin: <marker>Sonntag, 23.08.</marker></example>
-                <example correction="">Termin: <marker>Sonntag, 23.8.</marker></example>
-                <example correction="">Termin: <marker>So, 23.08.</marker></example>
-                <example correction="">Termin: <marker>So., 23.08.</marker></example>
-                <example correction="">Termin: <marker>Sonntag, den 23.08.</marker></example>
-                <example correction="">Termin: <marker>So, den 23.08.</marker></example>
-                <example correction="">Termin: <marker>So., den 23.08.</marker></example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>.08.</example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>.8.</example>
+                <example correction="Samstag, 23|So, 24">Termin: <marker>So, 23</marker>.08.</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30684]: --> So.,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 15]
```diff
@@ -30677,15 +30677,15 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
+                <example correction="Samstag., 23|So., 24">Termin: <marker>So., 23</marker>.08.</example>
+                <example correction="Samstag, den 23|Sonntag, den 24">Termin: <marker>Sonntag, den 23</marker>.08.</example>
+                <example correction="Samstag, den 23|So, den 24">Termin: <marker>So, den 23</marker>.08.</example>
+                <example correction="Samstag., den 23|So., den 24">Termin: <marker>So., den 23</marker>.08.</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30688]: --> Samstag,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 16]
```diff
@@ -30677,15 +30677,15 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
-                <example correction="">Termin: <marker>So., 23.08.</marker></example>
-                <example correction="">Termin: <marker>Sonntag, den 23.08.</marker></example>
-                <example correction="">Termin: <marker>So, den 23.08.</marker></example>
-                <example correction="">Termin: <marker>So., den 23.08.</marker></example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>.08.</example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>.8.</example>
+                <example correction="Samstag, 23|So, 24">Termin: <marker>So, 23</marker>.08.</example>
+                <example correction="Samstag., 23|So., 24">Termin: <marker>So., 23</marker>.08.</example>
+                <example correction="Samstag, den 23|Sonntag, den 24">Termin: <marker>Sonntag, den 23</marker>.08.</example>
+                <example correction="Samstag, den 23|So, den 24">Termin: <marker>So, den 23</marker>.08.</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30687]: --> So.,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 17]
```diff
@@ -30711,13 +30711,13 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
+                <filter class="org.languagetool.rules.de.DateCheckFilter" args="month:7 day:5 weekDay:1"/>
                 <message>Bezieht sich dieses Datum auf das aktuelle Jahr? Der \5. \7 {currentYear} fällt nicht auf einen {day}, sondern auf einen {realDay}.</message>
-                <example correction="">Termin: <marker>Donnerstag, 1. Jänner</marker></example>
-                <example correction="">Termin: <marker>Sonntag, 23. August</marker></example>
-                <example correction="">Termin: <marker>So, 23. August</marker></example>
-                <example correction="">Termin: <marker>So., 23. August</marker></example>
-                <example correction="">Termin: <marker>Sonntag, den 23. August</marker></example>
+                <example correction="Mittwoch, 1|Donnerstag, 2">Termin: <marker>Donnerstag, 1</marker>. Jänner</example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>. August</example>
+                <example correction="Samstag, 23|So, 24">Termin: <marker>So, 23</marker>. August</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30718]: --> So.,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 18]
```diff
@@ -30711,13 +30711,13 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
+                <example correction="Samstag., 23|So., 24">Termin: <marker>So., 23</marker>. August</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30719]: --> Samstag,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 19]
```diff
@@ -30733,12 +30733,12 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
-                <filter class="org.languagetool.rules.de.DateCheckFilter" args="month:\6 day:\4 weekDay:\1"/>
+                <filter class="org.languagetool.rules.de.DateCheckFilter" args="month:6 day:4 weekDay:1"/>
                 <message>Bezieht sich dieses Datum auf das aktuelle Jahr? Der \4. \6 {currentYear} fällt nicht auf einen {day}, sondern auf einen {realDay}.</message>
-                <example correction="">Termin: <marker>Donnerstag, 1. Jän.</marker></example>
-                <example correction="">Termin: <marker>Sonntag, 23. Aug.</marker></example>
-                <example correction="">Termin: <marker>So, 23. Aug.</marker></example>
-                <example correction="">Termin: <marker>Sonntag, den 23. Aug.</marker></example>
+                <example correction="Mittwoch, 1|Donnerstag, 2">Termin: <marker>Donnerstag, 1</marker>. Jän.</example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>. Aug.</example>
+                <example correction="Samstag, 23|So, 24">Termin: <marker>So, 23</marker>. Aug.</example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30740]: --> So.,

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml [hunk 20]
```diff
@@ -30733,12 +30733,12 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
...
+                <example correction="Samstag, den 23|Sonntag, den 24">Termin: <marker>Sonntag, den 23</marker>. Aug.</example>
                 <example>Termin: <marker>Mittwoch, 1. Jän.</marker></example>
                 <example>Termin: <marker>Samstag, 23. Aug.</marker></example>
                 <example>Termin: <marker>Sa, 23. Aug.</marker></example>
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30744]: Sa.,

**Review by Luisa-LT — COMMENTED** [review]
When the weekday is spelled out (Samstag), we don't need the dot (only the comma: Samstag,) but when the weekday is shortened (So.) we need the dot (and the comma: So.,)

File: languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/DateCheckFilter.java [hunk 21]
```diff
@@ -35,6 +36,11 @@ protected Calendar getCalendar() {
     return dateFilterHelper.getCalendar();
   }
 
+  @Override
+  protected String getErrorMessageWrongYear()  {
+    return "Dieses Datum stimmt nicht. Meinten Sie \"{currentYear}\"?";
```
Comment (Luisa-LT) [inline languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/DateCheckFilter.java:41]: Perhaps: "Dieses Datum stimmt nicht mit dem Tag überein." (This date does not coincide with the day.)

File: languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml (hunk 1 above)
Comment (jaumeortola) [reply languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml:30510]: Thank you. I think I can fix it, without making it too complicated.

**Pull request merged**